Fix fail to call lambda without arguments

carmenlau opened the pull request:
connects #276

This bug occurs after adding auto upload asset when calling lambda, sdk will check the arguments if there is any not updated assets.
Update the callLambdaFunction implementation to skip the presave.

Steven-Chan commented on skygear/src/main/java/io/skygear/skygear/Container.java:
```diff
@@ -203,7 +203,9 @@ public void sendRequest(Request request) {
      */
     public void callLambdaFunction(String name, LambdaResponseHandler handler) {
-        this.callLambdaFunction(name, (Object[]) null, handler);
+        LambdaRequest request = new LambdaRequest(name, (Object)null);
```
Seems update `public void callLambdaFunction(final String name, Object[] args, LambdaResponseHandler handler)` to handle using `null` for `Object[] args` is a more complete solution.

carmenlau commented:
@Steven-Chan @kaojohnny I updated the PR a bit. Both call lambda with array or map args should handle the null value case. 
Also I found that the `onFailure` of presave function hasn't called the `responseHandler.onFail`. I updated in the last commit too. 
Thanks

carmenlau commented:
Merged to master via fa6ec6c
